### PR TITLE
Add timeout support to PanMessaging.receive_message

### PR DESCRIPTION
**Pull request by jamessynge**
Add testing of conftest.py messaging fixtures to
test_messaging.py.

Delay message_forwarder fixture to allow forwarder to start.
Will later work on reducing that delay or eliminating the
need for a delay.

Remove unused imports.

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -50,7 +52,8 @@ def pytest_addoption(parser):
         "--test-databases",
         nargs="+",
         default=['mongo'],
-        help="Test databases in the list. Note that travis-ci will test all of them by default.")
+        help=("Test databases in the list. List items can include: " + db_names +
```
Comment (jamessynge) [inline conftest.py]: Sorry, I apparently couldn't resist adding an improvement to the help text while I was working in this file.

File: conftest.py [hunk 2]
```diff
@@ -267,6 +270,9 @@ def message_forwarder(messaging_ports):
         args.append(str(pub))
 
     proc = subprocess.Popen(args, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+    # It takes a while for the forwarder to start, so allow for that.
+    # TODO(jamessynge): Come up with a way to speed up these fixtures.
+    time.sleep(3)
```
Comment (wtgee) [inline conftest.py]: Were we missing messages in some of the tests?

File: pocs/utils/messaging.py [hunk 3]
```diff
@@ -201,7 +206,7 @@ def send_message(self, topic, message):
         # Send the message
         self.socket.send_string(full_message, flags=zmq.NOBLOCK)
 
-    def receive_message(self, blocking=True, flags=0):
+    def receive_message(self, blocking=True, flags=0, timeout_ms=0):
```
Comment (wtgee) [inline pocs/utils/messaging.py]: Can you add the param to the docstring.

File: pocs/utils/messaging.py [hunk 4]
```diff
@@ -217,9 +222,15 @@ def receive_message(self, blocking=True, flags=0):
...
+        elif timeout_ms > 0:
+            # Wait until a message is available or the timeout expires.
+            self.socket.poll(timeout=timeout_ms, flags=(zmq.POLLIN | zmq.POLLOUT))
+            # Don't block at this point, because we will have waited as long
+            # as necessary.
+            flags = flags | zmq.NOBLOCK
         try:
             message = self.socket.recv_string(flags=flags)
-        except Exception:
+        except Exception as e:
```
Comment (wtgee) [inline pocs/utils/messaging.py]: I'm surprised pyflakes doesn't complain about not using the `e`. Any reason it was added?

File: pocs/tests/test_messaging.py [hunk 5]
```diff
@@ -85,3 +85,60 @@ def test_storage_id(pub_and_sub, config, db):
...
+    topic, msg_obj = sub.receive_message()
+    assert isinstance(msg_obj, dict)
+    assert 'message' in msg_obj
+    assert msg_obj['message'] == 'a string'
+    assert 'timestamp' in msg_obj
+
+    # import pdb;pdb.set_trace()
+
+
+def test_msg_pub_sub(msg_publisher, msg_subscriber):
```
Comment (wtgee) [inline pocs/tests/test_messaging.py]: Seems like these two tests could be one but with a param of params. That said, this probably is more obvious.

File: conftest.py (hunk 2 above)
Comment (jamessynge) [reply conftest.py]: Only for tests that I'd written, and was trying to add in another PR that I'll send soon for coverage of the arduino i/o code. I'm going to try to improve the msg forwarding support in the tests in general a bit later.

File: pocs/utils/messaging.py (hunk 4 above)
Comment (jamessynge) [reply pocs/utils/messaging.py]: Good question. Probably because I was using the debugger. I'll double check pyflakes on this.

**Comment by jamessynge** [comment]
Adding this in support of testing. In particular, want to be able to call receive_message but not block indefinitely if something is wrong.

File: pocs/utils/messaging.py (hunk 3 above)
Comment (jamessynge) [reply pocs/utils/messaging.py]: Done.

File: pocs/tests/test_messaging.py (hunk 5 above)
Comment (jamessynge) [reply pocs/tests/test_messaging.py]: Ack.

**Comment by jamessynge** [comment]
The `create_forwarder_sockets` code currently runs in another process, and isn't captured by coverage AFAICT.

**Pull request merged**